platform/commonUI/general/src/ui/TreeNodeView.js

```diff
@@ -1,155 +1,164 @@
-                activeIdPath.every(function (id, index) {
-                    return selectedIdPath[index] === id;
-                });
+            function getIdPath(domainObject) {
+                var context = domainObject && domainObject.getCapability('context');
 
-        if (this.onSelectionPath) {
-            if (activeIdPath.length === selectedIdPath.length) {
-                this.li.find('.tree-item').eq(0).addClass('selected');
-            } else {
-                // Expand to reveal the selection
-                this.toggleView.value(true);
-                this.subtreeView.value(domainObject);
+                function getId(domainObj) {
+                    return domainObj.getId();
+                }
+
+                return context ? context.getPath().map(getId) : [];
             }
-        }
-    };
 
-    /**
-     *
-     * @returns {HTMLElement[]}
-     */
-    TreeNodeView.prototype.elements = function () {
-        return this.li;
-    };
+            TreeNodeView.prototype.value = function (domainObject) {
+                var activeIdPath = getIdPath(this.activeObject),
+                    selectedIdPath = getIdPath(domainObject);
+
+                if (this.onSelectionPath) {
+                    this.li.find('.tree-item').eq(0).removeClass('selected');
+                    if (this.subtreeView) {
+                        this.subtreeView.value(undefined);
+                    }
+                }
+
+                this.onSelectionPath =
+                    !!domainObject &&
+                    !!this.activeObject &&
+                    (activeIdPath.length <= selectedIdPath.length) &&
+                        activeIdPath.every(function (id, index) {
+                            return selectedIdPath[index] === id;
+                        });
+
+                if (this.onSelectionPath) {
+                    if (activeIdPath.length === selectedIdPath.length) {
+                        this.li.find('.tree-item').eq(0).addClass('selected');
+                    } else {
+                        // Expand to reveal the selection
+                        this.toggleView.value(true);
+                        this.subtreeView.value(domainObject);
+                    }
+                }
+            };
+
+            /**
+             *
+             * @returns {HTMLElement[]}
+             */
+            TreeNodeView.prototype.elements = function () {
+                return this.li;
+            };
+
+            TreeNodeView.prototype.hasChildren = function (domainObject) {
+
+                if (domainObject) {
+                    if (domainObject.hasCapability('composition')) {
+                        var proto = Object.getPrototypeOf(domainObject);
+                        var elementCount = proto.model.composition.length;
+                        if (elementCount > 0) {
+                            $(this.toggleView.elements()).addClass('has-children');
+                        }else {
```

Comment:
Missing tab

Reply:
Hi @ev1stensberg Can you point out where, I have just seen gulp checkstyle
My all checks are passing :)

Reply:
Sometimes the symlink with gulp is buggy, so you have to run the linting things yourself. Checkout the gulpfile in the root of the project :)